### PR TITLE
Use napari's QtListView for KeyFramesListWidget

### PR DESCRIPTION
**Pull request by tlambert03**
This PR follows on the work that @Fifourche is doing in #90 and swaps out the `KeyFramesListWidget` for napari's `QtListView`.

This PR _includes_ #90, so that PR should be merged first... but this gives a sense for how that PR can be leveraged to remove a big widget from the codebase.

I'm afraid I also committed the sin of touching unrelated things: specifically, I changed a number of the single-use methods like `_init_x()` to be used inline in the `__init__` method.  This is _entirely_ personal preference... so @alisterburt, just say the word and I'll revert those changed.  Even though it does make for a longer method, I find that it's _much_ easier to follow the flow of QtWidget creation if I don't have to go search what each `_init_something()` method is doing individually.  Also, it makes it a bit easier to see what order things are being added to the layout when they're all right next to each other.

### CODE REVIEW

**Review by alisterburt — APPROVED** [review]
This is fantastic @tlambert03 !! Your ruthless refactoring is very much appreciated 👍🏼 

I got import errors trying to use this with napari 0.4.7 but switching to 0.4.8 fixed that - do you think we should bump the requirement? I'm happy for it to move up to 0.4.8, if you are too would you mind doing that before merge?

There's a slight regression, introduced in #90 I assume - the bug fixed in #69 has come back, can be seen by creating a simple animation and moving the `AnimationSliderWidget` from right to left slowly - I'm not immediately sure of the best way to fix this with the `SelectableEventedList` but it should be possible! Happy to merge and get this fixed in a follow up, will open a separate issue

So satisfying to see all that red in the diff! 🙂

File: napari_animation/_qt/keyframeslist_widget.py
Comment (tlambert03) [reply napari_animation/_qt/keyframeslist_widget.py:7]: I don't have any particular convention to point to here :) I'd say whatever feels right.  I put them in the same file here since they're both small, and likely to always be used together.

File: napari_animation/_qt/keyframeslist_widget.py
Comment (tlambert03) [reply napari_animation/_qt/keyframeslist_widget.py:12]: yeah, the concept of "data roles" are a little weird at first.

File: napari_animation/utils.py
Comment (tlambert03) [reply napari_animation/utils.py:66]: lol :)  just making the individual keyframes a _little_ bit shorter in height.
but this would actually be a great feature to add: something that controls the size of each keyframe in the list

File: napari_animation/_qt/animation_widget.py
Comment (tlambert03) [reply napari_animation/_qt/animation_widget.py:62]: glad you agree.  I'd say the main exception to this would be if you find that you need to _re_-initialize things in the lifetime of the widget.  Then, it makes sense to pull that out into a method that can be called.

Comment (sofroniewn) [reply napari_animation/_qt/animation_widget.py:62]: Yeah, agreed i like it too, thanks

File: napari_animation/_qt/animation_widget.py
Comment (sofroniewn) [inline napari_animation/_qt/animation_widget.py:70]: hopefully one day we'll be able to use the new action manager stuff in napari here!!

**Review by sofroniewn — APPROVED** [review]
LGTM, looks like we just have some black/ flake errors, but apart from that this seems ready to go, nice work both!!

**Comment by tlambert03** [comment]
glad you like it.  don't merge this just yet.  I found a couple issues with `_current` vs `active` that I want to hammer out first (might actually need fixing upstream).

**Comment by tlambert03** [comment]
ok, figured it out and fixed it here.  There _is_ a slight issue that I'll open upstream, which is that when you _set_ an item in the list `key_frames[i] = new_frame` then the `removed` event isn't emitted for the previous item (instead a `changed` event is emitted... but that fails to remove the old item from the `selection` if it was selected before deletion).   So the replace-keyframe option wasn't working.  I do it in two steps now  (i.e. first remove the old item, then insert the new one), and will follow up in napari.  

Other than the regression of #69, this is ready to go.  Can fix that here or in a new PR

**Comment by tlambert03** [comment]
fixed the slider regression

**Comment by alisterburt** [comment]
amazing, thanks for this and fixing the regression too! let's get this merged 🚀

**Pull request merged**
